server_installation/topics/operating-mode/crossdc/proc-setting-up-infinispan.adoc

```diff
@@ -0,0 +1,172 @@
+[id='setting-up-infinispan-{context}']
+= Setting Up {jdgserver_name} Clusters
+For Cross-Datacenter replication, you start by creating remote {jdgserver_name} clusters that can back up {project_name} data.
+
+.Prerequisites
+
+* Download and install {jdgserver_name} Server {jdgserver_version_latest}.
+
+[NOTE]
+====
+{jdgserver_name} Server {jdgserver_version_latest} requires Java 11.
+====
+
+* Create a user to authenticate client connections from {jdgserver_name}, for example:
++
+[source,bash,options="nowrap",subs=attributes+]
+----
+$ bin/cli.sh user create myuser -p "qwer1234!"
+----
++
+You specify these credentials in the Hot Rod client configuration when you set up {jdgserver_name}.
+
+* Create an SSL keystore and truststore to secure connections between {jdgserver_name} and {project_name}.
+
+For example, you can use keytool as follows:
+
+[source,bash,options="nowrap",subs=attributes+]
+----
+# Create a keystore to provide an SSL identity to your {jdgserver_name} cluster
+keytool -genkey -alias server -keyalg RSA -keystore server.jks -keysize 2048
+
+# Export an SSL certificate from the keystore
+keytool -exportcert -keystore server.jks -alias server -file server.crt
+
+# Import the SSL certificate into a truststore that {jdgserver_name} can use to verify the SSL identity for {jdgserver_name}
+keytool -importcert -keystore truststore.jks -alias server -file server.crt
+rm server.crt
+----
+
+.Procedure
+
+. Open `infinispan.xml` for editing.
++
+By default, {jdgserver_name} Server uses `server/conf/infinispan.xml` for static configuration such as cluster transport and security mechanisms. 
+
+. Create a stack the uses TCCPING. The default tcp stack finds nodes by using IP multicast, which is typically unavailable.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<stack name="global-cluster" extends="tcp">
+    <!-- Remove MPING protocol from the stack and add TCPPING -->
+    <TCPPING initial_hosts="<ip1>[<port>],<ip2>[<port>]" stack.combine="REPLACE" stack.position="MPING"/>
```

Comment:
@mposolda If I look at the default configuration of Infinispan JGroups stacks I see that the TCP protocol is actually using port 7800 instead of 7600.
```
default-jgroups-tcp.xml (infinispan-core.jar):
...
   <TCP bind_addr="${jgroups.tcp.address:127.0.0.1}"
        bind_port="${jgroups.tcp.port:7800}"
        ...
```
https://infinispan.org/docs/dev/titles/server/server.html#jgroups_ports-cluster-transport

I have also found that I need to set these properties explicitely during startup (like so: `./server.sh ... -Djgroups.tcp.address=server1 -Djgroups.tcp.port=7800`), otherwise Infinispan won't pick them up, and the TCPPING doesn't work (the bridge cluster doesn't connect).

Reply:
In other words, do I need to use this syntax:   initial_hosts="server1[7800],server2[7800]"    NOT 7600?
Also, for setting the properties at startup, you have './server.sh -Djgroups.tcp.address=server1 -Djgroups.tcp.port=780"
Should that begin with ./standalone.sh  ?
Can you provide a full example, @tkyjovsk

Reply:
@tkyjovsk which infinispan-core.jar version are you checking? For Infinispan 10+, the config is a little different: https://github.com/infinispan/infinispan/blob/master/core/src/main/resources/default-configs/default-jgroups-tcp.xml#L5

> I have also found that I need to set these properties explicitly during startup (like so: ./server.sh ... -Djgroups.tcp.address=server1 -Djgroups.tcp.port=7800), otherwise Infinispan won't pick them up, and the TCPPING doesn't work (the bridge cluster doesn't connect).

can you elaborate? You don't have to set the properties explicitly if you are ok with the default value.

@andymunro yes to `initial_hosts="server1[7800],server2[7800]"`
Infinispan 10+ does not use WF anymore. The start script is `./bin/server.sh`

Reply:
@pruivo I was checking `11.0.3.Final` however the above snippet is actually from `9.4.19.Final` -- sorry for that. The infinispan-core-11.0.3 contains this:
```
   <TCP bind_addr="${jgroups.bind.address,jgroups.tcp.address:SITE_LOCAL}"
        bind_port="${jgroups.bind.port,jgroups.tcp.port:7800}"
        ...
```

The problem is that it shouldn't be necessary to provide the `jgroups.tcp.address` and `jgroups.tcp.port` via command line as they should default to predefined values. However if I leave them out the bridge cluster doesn't connect. I get this on site 1:
```
16:30:56,749 INFO  (jgroups-6,localhost-57801) [org.jgroups.protocols.relay.RELAY2] _localhost-57801:site1: joined bridge cluster 'global-cluster'
16:30:56,752 INFO  (jgroups-7,localhost-57801) [org.infinispan.XSITE] ISPN000439: Received new x-site view: [site1]
```
and this on site 2:
```
16:31:09,667 INFO  (jgroups-6,localhost-8675) [org.jgroups.protocols.relay.RELAY2] _localhost-8675:site2: joined bridge cluster 'global-cluster'
16:31:09,669 INFO  (jgroups-7,localhost-8675) [org.infinispan.XSITE] ISPN000439: Received new x-site view: [site2]
```
Instead of `[site1,site2]`